src/Dapr.Jobs/DaprJobHandler.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Dapr.Jobs.Models.Responses;
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// The delegate representing how a triggered job is handled.
+/// </summary>
+/// <param name="serviceProvider">The service provider that facilitates injecting resources to the delegate.</param>
+/// <param name="jobName">The name of the triggered job.</param>
+/// <param name="jobDetails">The details of the triggered job.</param>
+/// <returns></returns>
+public delegate Task InjectableDaprJobHandler(IServiceProvider serviceProvider, string? jobName, DaprJobDetails? jobDetails );
```

Comment:
A thought: there may be other things we'll want to pass to such handlers (e.g. a `CancellationToken`).  It might be best to consolidate the individual arguments into a `DaprJobHandlerContext` object and just pass that one thing.  In that way, additional things can be added in a way that won't break existing code and won't require additional overloads of the `MapDaprJob()` method.

It would be cool to be able to reuse some of ASP.NET's minimal API parameter binding logic so that users can just write their handlers in the same way as they normally would (while just injecting the additional job details), but I don't think that's exposed from the infrastructure.

Reply:
@philliphoff 

I've been trying to think of some way to make this more flexible and while reading through a little bit of how ASP.NET Core does it in their minimal APIs, I've come up with a similar approach and while it's not as fancy as theirs, it also doesn't need to do route binding. Ideally, we can add an analyzer for this or something at some point, but the developer can pass a delegate of their own into the extension and the first two parameters must be a `string?` for the job name and a `DaprJobDetails?` for the job details. Anything after that is dynamically injected for them like the minimal APIs do and the cancellationToken is passed in as the last parameter.

This approach gets rid of the two delegate handlers I defined and means the developer doesn't have to deal with an `IServiceProvider`, but gets their `CancellationToken` and we don't have to deal with maintaining a context object.

I want to write several unit tests to validate this out and make sure it works, but I updated the [example](https://github.com/dapr/dotnet-sdk/blob/10cc382c60c4f13a4edb898c4d5e628bd46a97d0/examples/Jobs/JobsSample/Program.cs#L16) to show off the new approach.